components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/utils/RESTServiceUtils.java

```diff
@@ -0,0 +1,478 @@
+import org.apache.axiom.om.OMNamespace;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.registry.core.Registry;
+import org.wso2.carbon.registry.core.RegistryConstants;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.ResourceImpl;
+import org.wso2.carbon.registry.core.config.RegistryContext;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.jdbc.handlers.RequestContext;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+import org.wso2.carbon.registry.extensions.utils.CommonConstants;
+
+import javax.xml.namespace.QName;
+import java.util.*;
+
+/**
+ * This class contains static methods to generate REST Service registry artifact from the swagger doc added to the
+ * Registry.
+ */
+public class RESTServiceUtils {
+
+	private static final Log log = LogFactory.getLog(RESTServiceUtils.class);
+	private static final String OVERVIEW = "overview";
+	private static final String PROVIDER = "provider";
+	private static final String NAME = "name";
+	private static final String CONTEXT = "context";
+	private static final String VERSION = "version";
+	private static final String TRANSPORTS = "transports";
+	private static final String DESCRIPTION = "description";
+	private static final String URI_TEMPLATE = "uritemplate";
+	private static final String URL_PATTERN = "urlPattern";
+	private static final String AUTH_TYPE = "authType";
+	private static final String HTTP_VERB = "httpVerb";
+	private static final String ENDPOINT_URL = "endpointURL";
+	private static final String WADL = "wadl";
+	private static final String PATH_SEPERATOR = "/";
+	private static final String METHOD = "method";
+	private static final String PATH = "path";
+	private static final String RESOURCE = "resource";
+
+	private static OMFactory factory = OMAbstractFactory.getOMFactory();
+	private static OMNamespace namespace = factory.createOMNamespace(CommonConstants.SERVICE_ELEMENT_NAMESPACE, "");
+	private static String commonRestServiceLocation;
+	private static String commonEndpointLocation;
+
+	/**
+	 * Extracts the data from swagger and creates an REST Service registry artifact.
+	 *
+	 * @param swaggerDocObject      swagger Json Object.
+	 * @param swaggerVersion        swagger version.
+	 * @param resourceObjects       swagger resource object list.
+	 * @return                      The API metadata
+	 * @throws RegistryException    If swagger content is invalid.
+	 */
+	public static OMElement createRestServiceArtifact(JsonObject swaggerDocObject, String swaggerVersion,
+	                                                  String endpointURL, List<JsonObject> resourceObjects)
+			throws RegistryException {
+
+		if(swaggerDocObject == null || swaggerVersion == null) {
+			throw new IllegalArgumentException("Arguments are invalid. cannot create the REST service artifact. ");
```

Comment:
@prasa7 Added a more elaborated message with argument context information in 818ebe20b0878325e9f7fddd1b47d83e86f7a34b